src/lib.rs

```diff
@@ -91,11 +104,19 @@ where
 /// ```
 pub fn totp_custom<H>(step: u64, digits: u32, secret: &[u8], time: u64) -> String
 where
-    H: Update + BlockInput + Reset + FixedOutputDirty + Clone + Default,
+    H: Update + FixedOutput + CoreProxy,
+    H::Core: HashMarker
+        + UpdateCore
+        + FixedOutputCore
+        + BufferKindUser<BufferKind = Eager>
+        + Default
+        + Clone,
+    <H::Core as BlockSizeUser>::BlockSize: IsLess<U256>,
+    Le<<H::Core as BlockSizeUser>::BlockSize, U256>: NonZero,
 {
     // Hash the secret and the time together.
-    let mut mac: Hmac<H> = Hmac::new_from_slice(secret).unwrap();
-    mac.update(&to_bytes(time / step));
+    let mut mac = <Hmac<H> as Mac>::new_from_slice(secret).unwrap();
+    <Hmac<H> as Update>::update(&mut mac, &to_bytes(time / step));
```

Comment:
Is `update` no longer a method?

Reply:
It is still a method, but both the `Update` and the `UpdateCore` traits implement `update`, so I had to let the compiler know which of the two implementations to use